cmd/iam-store.go

```diff
@@ -2170,6 +2185,15 @@ func (store *IAMStoreSys) UpdateServiceAccount(ctx context.Context, accessKey st
 		cr.Comment = opts.comment
 	}
 
+	if opts.expiration != nil {
```

Comment:
This seems to be one place we are preferring `*time.Time{}` over `time.Time{}` - @klauspost I saw the discussion in the related PRs (last one at https://github.com/minio/madmin-go/pull/175#discussion_r1108247453), I don't mind either way (and indeed I prefer using nil as a clear "not present" indication), but am wondering why standard lib documents a recommended usage like this:

```
Programs using times should typically store and pass them as values, not pointers. That is, time variables and struct fields should be of type time.Time, not *time.Time. 
```

at https://pkg.go.dev/time#Time

Reply:
@Praveenrajmani It would also make sense to add an upper bound on the expiration of 1 year, just to ensure that ridiculous values are not set by users (accidentally or not). 1 year is also the upper bound on STS credentials in the current implementation IIRC.

Reply:
thanks, like LDAP expiry, i'm setting the upper bound to 1 year and lower bound to 15 mins.